eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -766,16 +766,39 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
   }
 
   private def signTx(unsignedTx: SharedTransaction): Unit = {
+    import fr.acinq.bitcoin.scalacompat.KotlinUtils._
+
     val tx = unsignedTx.buildUnsignedTx()
     val sharedSig_opt = fundingParams.sharedInput_opt.map(_.sign(keyManager, channelParams, tx))
     if (unsignedTx.localInputs.isEmpty) {
       context.self ! SignTransactionResult(PartiallySignedSharedTransaction(unsignedTx, TxSignatures(fundingParams.channelId, tx, Nil, sharedSig_opt)))
     } else {
-      context.pipeToSelf(wallet.signTransaction(tx, allowIncomplete = true).map {
-        case SignTransactionResponse(signedTx, _) =>
+      // We only sign our wallet inputs, and check that we can spend our wallet outputs.
```

Comment:
AFAICT we don't "check that we can spend our wallet outputs.", we rather verify that the outputs that we have specified have been left untouched? In that case, why does it matter that we control them or not (change vs splice output)?

Reply:
Let me try to rephrase. I think we should:
- verify that all provided inputs and outputs have been preserved (entirely: amounts, scripts, etc.)
- verify that we control all new output scripts